fluss-rpc/src/main/java/com/alibaba/fluss/rpc/netty/NettyMemoryMetrics.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2024 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.rpc.netty;
+
+import com.alibaba.fluss.metrics.Gauge;
+import com.alibaba.fluss.metrics.groups.MetricGroup;
+import com.alibaba.fluss.shaded.netty4.io.netty.buffer.PoolArenaMetric;
+import com.alibaba.fluss.shaded.netty4.io.netty.buffer.PooledByteBufAllocator;
+import com.alibaba.fluss.shaded.netty4.io.netty.buffer.PooledByteBufAllocatorMetric;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/** A Netty memory metrics class to collect metrics from Netty {@link PooledByteBufAllocator}. */
+public class NettyMemoryMetrics {
+
+    private final MetricGroup metricGroup;
+    private final PooledByteBufAllocator pooledAllocator;
+    private final boolean verboseMetricsEnabled;
+
+    public static final String NETTY_MEMORY_METRIC_GROUP = "nettyMemory";
+    public static final String NETTY_MEMORY_USED_HEAP_MEMORY = "usedHeapMemory";
+    public static final String NETTY_MEMORY_USED_DIRECT_MEMORY = "usedDirectMemory";
+    public static final String NETTY_MEMORY_USED_HEAP_ARENAS = "numHeapArenas";
+    public static final String NETTY_MEMORY_USED_DIRECT_ARENAS = "numDirectArenas";
+    public static final String NETTY_MEMORY_TINY_CACHE_SIZE = "tinyCacheSize";
+    public static final String NETTY_MEMORY_SMALL_CACHE_SIZE = "smallCacheSize";
+    public static final String NETTY_MEMORY_NORMAL_CACHE_SIZE = "normalCacheSize";
+    public static final String NETTY_MEMORY_NUM_THREAD_LOCAL_CACHES = "numThreadLocalCaches";
+    public static final String NETTY_MEMORY_CHUNK_SIZE = "chunkSize";
+    public static final String NETTY_MEMORY_HEAP_ARENA = "heapArena";
+    public static final String NETTY_MEMORY_DIRECT_ARENA = "directArena";
```

Comment:
Let's move all the metric names to `MetricNames` where we maintain all the metric names centralized.